### PR TITLE
Basic game setup, currency & gamedata

### LINKED ISSUE #2: Python Arcade Setup
## Feature

- Setup the base python arcade game
- 720p
- Windowed
- Scalable

### LINKED ISSUE #4: Currency
- Carried over to next wave
- Has a hard limit (999999)

## Type
- Gold
- Research

## Gold
- Used to purchase towers, upgrade towers

## Research
- Used to unlock tower upgrades

### LINKED ISSUE #12: Save Data Optional Features
## Feature

- Add persistent save data
- Must be able to save and load

### PR DESCRIPTION
**Pull request by gamerssassin**
### Feature
- Add `Resource` Parent Class
- Add `Gold`, `Research` Child Classes
- Add base game 
- Add basic gamedata saving & loading

### Misc
- Add Game run command in README.md 

## Additional context

Close #2
Close #4
Close #12

## Checklist
- [X] My code follows the code style of this project [How to use `black` autoformatter](https://github.com/psf/black#usage).
- [X] I have applied relevant labels and linked the PR to the relevant Issue.

### CODE REVIEW

**Comment by Jeb4dev** [comment]
Good job, let's remove interfaces, they are over complicating stuff, with only one week, we don't rly need them.

File: run_game.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import sys
+
+from src.main import main
+
+# The major, minor version numbers your require
+MIN_VER = (3, 9)
```
Comment (Jeb4dev) [inline run_game.py:6]: can the game run on older versions, 3.7 or 3.6 even? Should use min version it works

**Review by Jeb4dev — DISMISSED** [review]
Good job, let's remove interfaces, but you can do it after so push to main, sure

File: run_game.py (hunk 1 above)
Comment (MrTanoshii) [reply run_game.py:6]: Last PyWeek, there were issues on 3.8 (Sound if I recall correctly)

**Review by IrrationalBoolean — APPROVED** [review]

**Pull request merged**